unix: Fail gracefully when calling uv_shutdown twice

Keno opened the pull request:
Before this, one of the requests was left as a ghost in the system. See https://github.com/JuliaLang/julia/issues/5793

Nodejs-Jenkins commented:
Thank you for contributing this pull request! Here are a few pointers to make sure your submission will be considered for inclusion.

Commit loladiro/libuv@fda80e11fef100f397425ab93b8650fc83f206bf has the following error(s):
- Commit message must indicate the subsystem this commit changes
- Commit message line too long: 2

The following commiters were not found in the CLA:
- Keno Fischer

You can fix all these things _without_ opening another issue.

Please see [CONTRIBUTING.md](https://github.com/joyent/node/blob/master/CONTRIBUTING.md) for more information

indutny commented:
Wow, thanks for catching it! May I ask you to write a test for it? (and also fix all issues mentioned by @Nodejs-Jenkins )

Keno committed:
unix: Correct error when calling uv_shutdown twice

Before this one of the requests was left as a ghost in the system. See https://github.com/JuliaLang/julia/issues/5793

Keno commented:
Updated.

saghul commented on test/test-shutdown-twice.c:
```diff
@@ -0,0 +1,85 @@
+                     &h,
+                     (const struct sockaddr*) &addr,
+                     connect_cb);
+  printf("%d\n", r);
```
Debug leftover?

saghul commented:
LGTM. If you don't mind please remove the printf, otherwise I can remove it when merging.

saghul commented:
Thanks @loladiro, landed in https://github.com/joyent/libuv/commit/a284b90c03735095763523a7c63f80140dc6a03a

saghul commented:
For the record, I made a backport for 0.10 and landed it in https://github.com/joyent/libuv/commit/6e2021ca111a44449932792e0cb578e63705c2a2
